### PR TITLE
Add Kafka Event

### PR DESCRIPTION
**Pull request by prajesh1**
*Description of changes:*

Adding Kafka Event

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java:24]: `@EqualsAndHashCode` is redundant when using `@Data` - ref. https://projectlombok.org/features/Data

This also applies to sub-classes below

Comment (carlzogh) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java:24]: Could you also please add `@AllArgsConstructor` and `@Builder(setterPrefix = "with")` to this model and its sub-classes?

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        Map<String, List<KafkaEventRecord>> records = new HashMap<>();
+        for(Map.Entry<KafkaEvent.TopicPartition, List<KafkaEvent.KafkaEventRecord>> entrySet: eventRecords.entrySet()) {
+            records.put(entrySet.getKey().toString(), entrySet.getValue());
+        }
+        this.records = records;
+    }
+
+    public String getEventSource () {
+        return eventSource;
+    }
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java:48]: No need for the explicit getter, Lombok's `@Data` will generate it

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@EqualsAndHashCode
+/** Represents a Kafka Event. **/
+public class KafkaEvent {
+    private Map<String, List<KafkaEventRecord>> records;
+    public  String eventSource;
+    private String eventSourceArn;
+
+    public Map<String, List<KafkaEventRecord>> getRecords() {
+        return records;
+    }
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java:33]: No need for the explicit getter, Lombok's `@Data` will generate it

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.List;
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode
+/** Represents a Kafka Event. **/
+public class KafkaEvent {
+    private Map<String, List<KafkaEventRecord>> records;
+    public  String eventSource;
```
Comment (msailes) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java:28]: Can this be private?

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java (hunk 1 above)
Comment (prajesh1) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java]: Done

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java (hunk 3 above)
Comment (prajesh1) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java]: Done

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java (hunk 4 above)
Comment (prajesh1) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/KafkaEvent.java]: yes, thanks

**Pull request merged**